project/GenSodiumPlugin.scala

```diff
@@ -13,12 +13,23 @@ object GenSodiumPlugin extends sbt.AutoPlugin {
 
   override def buildSettings = Seq(
     /* See plugins.sbt for why this is dynamically loaded */
-    gensodium := {
-      getClass.getClassLoader
-        .loadClass("tsec.build.GenSodium")
-        .getMethod("main", classOf[Array[String]])
-        .invoke(null, Array.empty[String])
+    gensodium := Def.task {
+      if (canWeUseComSunTools_?) {
+        getClass.getClassLoader
+          .loadClass("tsec.build.GenSodium")
+          .getMethod("main", classOf[Array[String]])
+          .invoke(null, Array.empty[String])
+      } else {
+        sys.error("sodium \"macro\" not supported on your superior JDK, sorry")
+      }
     }
   )
 
+  lazy val canWeUseComSunTools_? = ( // yes, iff ...
+       // we're not in IntelliJ
+       !sys.env.getOrElse("XPC_SERVICE_NAME", "").toLowerCase.contains("intellij")
+       // and we're on oracle JDK (sorry Mr Stallman)
+    && sys.props.get("java.vendor").exists(_.toLowerCase.contains("oracle"))
```

Comment:
I couldn't figure out, if this is generec to distributions:  _java.vendor_, check https://stackoverflow.com/questions/32570116/whats-the-difference-between-java-vendor-vs-java-vm-vendor

Reply:
Currently I'm exploring _nixos_ and I receive _segfaults_. I try to figure out this today

Reply:
I may have a better idea

Reply:
I had a better idea. try again? otherwise I'm just going to convert this to a script and make people use (gasp!) `bash` or something predictable.

Reply:
@hrhino I thing my issue is related to nixos, I will try to come up with a `nix` file